File: backend/app/schemas/share.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from uuid import UUID


class ShareBase(BaseModel):
    content_type: str
    content_id: UUID
    platform: str


class ShareCreate(ShareBase):
    pass


class ShareResponse(ShareBase):
    id: UUID
    user_id: UUID
    share_count: int
    last_shared_at: datetime
    
    class Config:
        from_attributes = True


class ShareStats(BaseModel):
    content_type: str
    content_id: UUID
    total_shares: int
    platform_stats: dict


class ShareUrlResponse(BaseModel):
    url: str
    title: str
    description: Optional[str] = None
    image: Optional[str] = None

File: frontend/src/api/share.ts
import apiClient from './client';

export interface ShareData {
  content_type: string;
  content_id: string;
  platform: string;
}

export interface ShareStats {
  content_type: string;
  content_id: string;
  total_shares: number;
  platform_stats: {
    link: number;
    wechat: number;
    weibo: number;
    twitter: number;
    facebook: number;
    copy: number;
  };
}

export interface ShareUrlResponse {
  url: string;
  title: string;
  description?: string;
  image?: string;
}

export const shareApi = {
  // 记录分享
  recordShare: async (data: ShareData): Promise<void> => {
    return apiClient.post('/share', data);
  },

  // 获取分享统计
  getShareStats: async (contentType: string, contentId: string): Promise<ShareStats> => {
    return apiClient.get(`/share/stats/${contentType}/${contentId}`);
  },

  // 获取分享链接
  getShareUrl: async (contentType: string, contentId: string): Promise<ShareUrlResponse> => {
    return apiClient.get(`/share/url/${contentType}/${contentId}`);
  },

  // 记录分享并获取链接
  recordAndGetUrl: async (
    contentType: string,
    contentId: string,
    platform: string = 'link'
  ): Promise<ShareUrlResponse> => {
    return apiClient.post(`/share/record/${contentType}/${contentId}?platform=${platform}`);
  },
};